esrally/mechanic/telemetry.py

```diff
@@ -481,107 +481,90 @@ def record(self):
         current_sample = self.sample()
         for node_stats in current_sample:
             node_name = node_stats["name"]
+            collected_node_stats = {}
+
             if self.include_indices:
-                self.record_indices_stats(node_name, node_stats,
-                                          include=["docs", "store", "indexing", "search", "merges", "query_cache", "fielddata",
-                                                   "segments", "translog", "request_cache"])
+                collected_node_stats.update(
+                    self.collect_indices_stats(node_name, node_stats,
+                                               include=["docs", "store", "indexing", "search", "merges", "query_cache", "fielddata",
+                                                        "segments", "translog", "request_cache"]))
             if self.include_thread_pools:
-                self.record_thread_pool_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_thread_pool_stats(node_name, node_stats))
             if self.include_breakers:
-                self.record_circuit_breaker_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_circuit_breaker_stats(node_name, node_stats))
             if self.include_buffer_pools:
-                self.record_jvm_buffer_pool_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_jvm_buffer_pool_stats(node_name, node_stats))
             if self.include_mem_stats:
-                self.record_jvm_mem_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_jvm_mem_stats(node_name, node_stats))
             if self.include_network:
-                self.record_network_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_network_stats(node_name, node_stats))
             if self.include_process:
-                self.record_process_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_process_stats(node_name, node_stats))
+
+            self.metrics_store.put_doc(collected_node_stats,
+                                       level=MetaInfoScope.node,
+                                       node_name=node_name,
+                                       meta_data=self.metrics_store_meta_data)
 
         time.sleep(self.sample_interval)
 
-    def record_indices_stats(self, node_name, node_stats, include):
+    def flatten_stats_fields(self, prefix=None, stats=None, include=None, level=0):
```

Comment:
IMHO this method is pretty complex and does more than just flattening fields:

1. It flattens
2. It filters top-level keys
3. It formats the flattened keys

Can we separate those responsibilities so it is simpler to understand?